WakaTime/Watcher.swift

```diff
@@ -111,11 +111,13 @@ class Watcher: NSObject {
             observer.addToRunLoop()
             self.observer = observer
             self.observingElement = element
-            let activeWindow = element.getValue(for: kAXFocusedWindowAttribute) as! AXUIElement
-            if let currentPath = getCurrentPath(window: activeWindow, refcon: this) {
-                self.documentPath = currentPath
+            if let activeWindow = element.getValue(for: kAXFocusedWindowAttribute) {
```

Comment:
I'm still thinking of a guarding approach that fits here better.
```suggestion
            if let activeWindow = element.getValue(for: kAXFocusedWindowAttribute) as? AXUIElement {
```

And drop the next force-casting line.

I'm sure like you, it will return an `AXUIElement` always, but I still can't believe 100% Apple on this 😂

Reply:
Thank you for the suggested fix! The issue with this is that SwiftC will throw a warning that the conditional downcast to AXUIElement will always succeed. See https://stackoverflow.com/questions/43927167/trouble-retrieving-a-cgcolor-from-a-swift-dictionary for an explanation.

Reply:
Hahaha, Apple being Apple. Create a problem, and create some compiler stuff that auto-magically fixes it.
Thanks for the clarification; ignore my comment, please!

Reply:
Exactly 😂 Swift is indeed pretty weird when it comes to anything with `Any` or `AnyObject`. I have a personal method `unwrap()` that can use reflection to determine if an `Any` is actually `nil` for logging purposes. I am afraid the only way to really improve beyond what we have in WakaTime right now is implement the StackOverflow approach, but then again that's not really required, because `getValue()` has the AX-specific guard there.